internal/users/manager.go

```diff
@@ -184,6 +195,9 @@ func (m *Manager) UpdateUser(u types.UserInfo) (err error) {
 			defer cleanup()
 
 			g.GID = &gid
+		} else {
+			// The group already exists in the database, use the existing GID to avoid permission issues.
+			g.GID = &oldGroup.GID
```

Comment:
You can also keep this as the default case no?

E.g. setting it always to it and if we've a no found error, we do the if clausule...

Reply:
That would also change the GID if there's an error (for example when registering the group in line 190), which would be an unexpected side effect of this function.

Reply:
We're doing that anyways, no? I mean this logic:

```go
		oldGroup, err := m.findGroup(g)
		if err != nil && !errors.Is(err, cache.NoDataFoundError{}) {
			// Unexpected error
			return err
		}

		// The group may already exists in the database, use the existing GID to avoid permission issues.
		g.GID = &oldGroup.GID

		if errors.Is(err, cache.NoDataFoundError{}) {
			...
		}
```

It could be also set by default though, since in case of error we're returning with no consequences anyways, no?

Reply:
In the `if errors.Is(err, cache.NoDataFoundError{}) {` block there are these lines:

```
			gid, cleanup, err := m.temporaryRecords.RegisterGroup(g.Name)
			if err != nil {
				return fmt.Errorf("could not generate GID for group %q: %v", g.Name, err)
			}
```

If we return an error there, we shouldn't set the GID of the group to zero.

Also, this comment becomes incorrect (and the code harder to follow IMO):
```
		// The group may already exists in the database, use the existing GID to avoid permission issues.
		g.GID = &oldGroup.GID
```

Reply:
I see,